sql-plugin/src/main/java/com/nvidia/spark/rapids/InternalRowToColumnarBatchIterator.java

```diff
@@ -92,6 +91,19 @@ protected InternalRowToColumnarBatchIterator(
     this.numOutputBatches = numOutputBatches;
   }
 
+  private int calcNumRowsEstimate(long targetBytes) {
+    return (int) Math.max(1,
+        Math.min(Integer.MAX_VALUE - 1, targetBytes / sizePerRowEstimate));
```

Comment:
nit: casting is forced by  `targetBytes / sizePerRowEstimate`? I'd prefer pushing it down to the origin:

```suggestion
    return Math.max(1,
        Math.min(Integer.MAX_VALUE - 1, (int) (targetBytes / sizePerRowEstimate)));
```